lib/grape/endpoint.rb

```diff
@@ -245,33 +245,38 @@ def run
         @request = Grape::Request.new(env, build_params_with: namespace_inheritable(:build_params_with))
         @params = @request.params
         @headers = @request.headers
+        begin
+          cookies.read(@request)
+          self.class.run_before_each(self)
+          run_filters befores, :before
+
+          if (allowed_methods = env[Grape::Env::GRAPE_ALLOWED_METHODS])
+            raise Grape::Exceptions::MethodNotAllowed, header.merge('Allow' => allowed_methods) unless options?
+            header 'Allow', allowed_methods
+            response_object = ''
+            status 204
+          else
+            run_filters before_validations, :before_validation
+            run_validators validations, request
+            remove_aliased_params
+            run_filters after_validations, :after_validation
+            response_object = @block ? @block.call(self) : nil
+          end
 
-        cookies.read(@request)
-        self.class.run_before_each(self)
-        run_filters befores, :before
-
-        if (allowed_methods = env[Grape::Env::GRAPE_ALLOWED_METHODS])
-          raise Grape::Exceptions::MethodNotAllowed, header.merge('Allow' => allowed_methods) unless options?
-          header 'Allow', allowed_methods
-          response_object = ''
-          status 204
-        else
-          run_filters before_validations, :before_validation
-          run_validators validations, request
-          remove_aliased_params
-          run_filters after_validations, :after_validation
-          response_object = @block ? @block.call(self) : nil
-        end
+          run_filters afters, :after
+          cookies.write(header)
 
-        run_filters afters, :after
-        cookies.write(header)
+          # status verifies body presence when DELETE
+          @body ||= response_object
 
-        # status verifies body presence when DELETE
-        @body ||= response_object
+          # The Body commonly is an Array of Strings, the application instance itself, or a File-like object
+          response_object = file || [body]
 
-        # The Body commonly is an Array of Strings, the application instance itself, or a File-like object
-        response_object = file || [body]
-        [status, header, response_object]
+          [status, header, response_object]
+        ensure
+          # An error on the finallies should not prevent a succesful request from executing
+          begin run_filters finallies, :finally end
```

Comment:
`begin/end` looks useless. do you mean 
```ruby
run_filters finallies, :finally rescue nil
```
if so, it is certainly a bad practice.

Reply:
I did mean that. 
And I'd tend to agree is bad practice, however I would be concern about a `finally` breaking an otherwise successful request. 
Do you have an alternative suggestion? Or would you just allow the broken finally to break the request (There's certainly an argument for that)

Reply:
In ruby, when an `ensure` block fails, it just re-raises an exception. I definitely don't think we should be swallowing an error here. I expect to at least get a 500 back with the exception details.

Reply:
I agree with @dblock, this should be consistent with `ensure`.

a user should decide if he uses "safe" code in `finally` or he needs some additional guard.
otherwise some unexpected `NameError` in recently-added "finally" callback could stop the running all other "finallies" and no one will know about this.